src/terraform/command_helpers.go

```diff
@@ -14,79 +14,79 @@ import (
 
 // CompileVarFlags compiles the variable flags passed into a Terraform command
 func CompileVarFlags(deployConfig deploy.Config, secrets types.Secrets, imagesMap map[string]string) ([]string, error) {
-	vars := compileSecrets(secrets)
-	imageVars := compileDockerImageVars(deployConfig, imagesMap)
-	vars = append(vars, imageVars...)
-	envVars, err := compileServiceEnvVars(deployConfig, secrets)
+	varFlagsMap := CompileDockerImageVars(deployConfig, imagesMap)
+	envVars, err := CompileServiceEnvVars(deployConfig, secrets)
 	if err != nil {
 		return []string{}, errors.Wrap(err, "cannot compile service environment variables")
 	}
-	vars = append(vars, envVars...)
-	dependencyVars, err := compileDependencyVars(deployConfig)
+	util.Merge(varFlagsMap, envVars)
+	dependencyVars, err := CompileDependencyVars(deployConfig)
 	if err != nil {
 		return []string{}, errors.Wrap(err, "cannot compile dependency variables")
 	}
-	vars = append(vars, dependencyVars...)
-	return append(vars, "-var", fmt.Sprintf("aws_profile=%s", deployConfig.AwsConfig.Profile)), nil
+	util.Merge(varFlagsMap, dependencyVars)
+	util.Merge(varFlagsMap, map[string]string{"aws_profile": deployConfig.AwsConfig.Profile})
```

Comment:
I don't think its worth using util.Merge here and instead we can use:
`varFlagsMap["aws_profile"] = deployConfig.AwsConfig.Profile`